src/libraries/System.Transactions.Local/src/System/Transactions/DtcProxyShim/DtcProxyShimFactory.cs

```diff
@@ -47,7 +47,28 @@ internal DtcProxyShimFactory(EventWaitHandle notificationEventHandle)
         object? pvConfigPararms,
         [MarshalAs(UnmanagedType.Interface)] out ITransactionDispenser ppvObject);
 
-    [UnconditionalSuppressMessage("Trimming", "IL2050", Justification = "Leave me alone")]
+    [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2050:COMMarshalling",
```

Comment:
I do not think this is sufficient to guarantee that Com works.

The built-in Com marshalling also depends on a bunch of internal runtime helper types that may get stripped.

Reply:
The rest of the change is fine. I think this suppression should be deleted and we should leave the warning on.

Reply:
We should mark this `RequiresUnreferencedCode` with a message saying the current implementation is not compatible with trimming.

Reply:
I've updated the code to leave the warning in the library. If an app uses this code while trimming they will get a warning.